setup.py

```diff
@@ -57,6 +57,7 @@
     install_requires=[
         "ply",
         "future",
+        "six",
```

Comment:
I tried to avoid using `six` in the earlier work to make katcp Python 2 and 3 compatible. I didn't want to use two different libraries (future and six) for the compatibility.  Was that a reasonable thought?

I'm sure you spotted `future.utils.tobytes`, which is almost what we want, except it uses `latin-1` instead of `utf-8` encoding.

What about adding something like this to the `katcp.compat` module?
```python
def ensure_byte_str(value):
    """Coerce unicode string to bytes (UTF-8 encoding), no change if input is bytes."""
    if is_bytes(value):
        return value
    elif is_text(value):
        return value.encode('utf-8')
    else:
        raise TypeError(
                "Invalid type for string conversion: {}".format(type(value)))
```

Reply:
cool, I will future. I used six as it was already installed on the docker container.

Reply:
I was thinking of using `https://python-future.org/reference.html?highlight=to_bytes#future.utils.native_str_to_bytes` but let me rather add the above code to the `compat.py`

Reply:
Oh, yes.  `native_str_to_bytes` could also work - I didn't see that one.  As long as we are sure that the caller is using native strings.  That seems pretty reasonable, since it is the most convenient form.

Reply:
Was gonna suggest the same :-)